storage/migrate.go

```diff
@@ -214,12 +214,20 @@ func (d *Database) MigrateSchemaTo(ctx context.Context, target model.Version) er
 		return xerrors.Errorf("acquiring schema lock: %w", err)
 	}
 
+	// Remember to release the lock
+	defer func() {
+		err := SchemaLock.UnlockExclusive(ctx, db)
+		if err != nil {
+			log.Errorf("failed to release exclusive lock: %v", err)
+		}
+	}()
+
 	if err := initDatabaseSchema(ctx, db, d.SchemaConfig()); err != nil {
 		return xerrors.Errorf("initializing schema version tables: %w", err)
 	}
 
 	// Check if we need to create the base schema
-	if dbVersion.Patch == 0 {
+	if dbVersion.Major == 0 {
```

Comment:
This is not correct. Major is the version of the schema in use. Each major version will have its own base schema on which patches are applied.

Reply:
I followed the reproduction steps in #706 and observed the base schema to be erroneously reapplied, this is the 🐛. I believe the issue here is using a lack of patch versions to determine if the database is initialized, which is wrong since the base schema can exist without patch versions (as it does going from v0.7.5 -> beyond). I think the fix here is the following:
```suggestion
if !initialized {
```

`initialized` is true iff the Database has the version tables **and** they are populated, meaning the base schema has been applied.